### PR TITLE
update yum repo when build image

### PR DESCRIPTION
**Pull request by navyxliu**
*Issue #, if available:*
#57 

*Description of changes:*
Update installed packages. it guarantees that we won't derive outdated packages from base images. 
We did it before yum clean, so the whole yum cache wipe out before the docker image is finalized.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AzeemJiva — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -24,6 +24,7 @@ RUN set -eux \
     && grep -q '^gpgcheck=1' /etc/yum.repos.d/corretto.repo \
     && yum install -y java-1.8.0-amazon-corretto-devel-$version \
     && yum install -y fontconfig \
+    && yum update -y \
     && yum clean all
 
-ENV JAVA_HOME=/usr/lib/jvm/java-1.8.0-amazon-corretto
```
Comment (TianminShi) [inline Dockerfile:29]: is it intentional to remove the newline at the end ?

Comment (navyxliu) [reply Dockerfile:29]: Technically speaking, we always need to a newline to the end of all Linux files.
My editor (VIM) inserted it.   yes, it's intentional.

Comment (TianminShi) [reply Dockerfile:29]: sorry, I thought the newline was removed. but you are right and LGTM

**Comment by navyxliu** [comment]
docker image does bloat after this change, which is bad. 

amazoncorretto8-before                                            latest              23b1b4700dd7        3 minutes ago       389MB
amazoncorretto8-after                                             latest              42e22e6f8a7e        10 days ago         521MB

is there a workaround? @alvdavi

**Comment by Denusdv** [comment]
Hi Guys. do you plan to merge this fix into corretto:8-al2-full branch, I'd like to try it out, thanks

**Comment by navyxliu** [comment]
> Hi Guys. do you plan to merge this fix into corretto:8-al2-full branch, I'd like to try it out, thanks
hi, @Denusdv 

We won't merge this PR. It's because the PR will make the image size about 50% bigger.
The best practice is to update the base image. We have done that.  Here is the update of AL2.
https://github.com/docker-library/official-images/pull/7719

The security issues have solved.

**Pull request closed without merging**